hypothesis-python/tests/numpy/test_gen_data.py

```diff
@@ -300,3 +302,9 @@ def test_all_inferred_scalar_strategies_roundtrip(data, dtype):
     assume(ex == ex)  # If not, the roundtrip test *should* fail!  (eg NaN)
     arr[0] = ex
     assert arr[0] == ex
+
+
+@checks_deprecated_behaviour
+@given(nps.arrays(dtype='int8', shape=(1,), elements=st.just(300)))
```

Comment:
the overflow behaviour of floats and ints is sufficiently different that it would probably be worthwhile having a test for `float8` too

Reply:
Numpy doesn't support `float8` 😉 

But I agree that float tests would be good; to check that `dtype='float16'` isn't deprecated and to check that `dtype='float16', elements=st.floats()` *is*.

Reply:
Oops, sorry, I of course meant `float16`. `float8` would be... interesting.

Reply:
https://en.m.wikipedia.org/wiki/Minifloat

"minifloats are floating-point values represented with very few bits. Predictably, they are not well suited for general-purpose numerical calculations" - and note the "all values" table!